src/libpriv/rpmostree-core.c

```diff
@@ -2904,10 +2933,12 @@ checkout_package (OstreeRepo   *repo,
   /* Used in the no-rofiles-fuse path */
   opts.force_copy_zerosized = force_copy_zerosized;
 
-  if (files_skip && g_hash_table_size (files_skip) > 0)
+  if ((filter_data->files_skip && g_hash_table_size (filter_data->files_skip) > 0) ||
```

Comment:
This function is also used by the relabeling code. In that path, it passes `NULL` for what is now the `filter_data` argument. So you need to also check if `filter_data` itself is `NULL` before dereferencing here.

Reply:
:open_mouth: 
I think this may be it...

Reply:
That was the issue! Actually, I remember trying with just `filter_data->files_skip` as the if condition (also without first checking whether filter_data itself is NULL) and the test ran with no seg faults... oh, nvm, I see why that would've seemed like it works.